rwengine/src/render/TextRenderer.cpp

```diff
@@ -85,46 +89,92 @@ struct TextVertex {
-    }
+const std::vector<std::uint8_t> fontWidthsArial = {
+    27, 25, 55, 43, 47, 65, 53, 19, 29, 31, 21, 45, 23, 35, 27, 29, //  1
+    47, 33, 45, 43, 49, 47, 47, 41, 47, 45, 25, 23, 53, 43, 53, 39, //  2
+    61, 53, 51, 47, 49, 45, 43, 49, 53, 23, 41, 53, 45, 59, 53, 51, //  3
+    47, 51, 49, 49, 45, 51, 49, 59, 59, 47, 51, 31, 27, 31, 29, 27, //  4
+    19, 43, 45, 43, 43, 45, 27, 45, 43, 21, 33, 45, 23, 65, 43, 43, //  5
+    47, 45, 33, 41, 29, 43, 41, 61, 51, 43, 43, 67, 53, 67, 67, 71, //  6
+    53, 53, 53, 53, 65, 49, 45, 45, 45, 45, 23, 23, 23, 23, 51, 51, //  7
+    51, 51, 51, 51, 51, 51, 51, 43, 43, 43, 43, 65, 43, 45, 45, 45, //  8
+    45, 21, 21, 21, 21, 43, 43, 43, 43, 43, 43, 43, 43, 53, 43, 39, //  9
+    25, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, // 10
+    19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 11, 19, 19, // 11
+    19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, 19, // 12
+    19,
 
-    // Assumes contiguous a-z character encoding
-    for (char g = 0; g <= ('z' - 'a'); g++) {
-        glyphData[charToIndex('a' + g)].widthFrac = 0.7f;
-        glyphData[charToIndex('A' + g)].widthFrac = 0.7f;
-    }
-    // case 'i':
-    glyphData[charToIndex('i')].widthFrac = 0.4f;
-    glyphData[charToIndex('I')].widthFrac = 0.4f;
-    // case 'l':
-    glyphData[charToIndex('l')].widthFrac = 0.5f;
-    glyphData[charToIndex('L')].widthFrac = 0.5f;
-    // case 'm':
-    glyphData[charToIndex('m')].widthFrac = 1.0f;
-    glyphData[charToIndex('M')].widthFrac = 1.0f;
-    // case 'w':
-    glyphData[charToIndex('w')].widthFrac = 1.0f;
-    glyphData[charToIndex('W')].widthFrac = 1.0f;
-    // case 'accent aigu'
-    glyphData[0x91].widthFrac = 0.6f;
-}
+};
 
-void TextRenderer::setFontTexture(int index, const std::string& texture) {
-    if (index < GAME_FONTS) {
-        fonts[index] = texture;
+void TextRenderer::setFontTexture(font_t font, const std::string& textureName) {
+    if (font >= FONTS_COUNT) {
+        RW_ERROR("Illegal font: " << font);
+        return;
+    }
+    auto ftexture = renderer->getData()->findSlotTexture("fonts", textureName);
+    const glm::u32vec2 textureSize = ftexture->getSize();
+    glm::u8vec2 glyphOffset{textureSize.x / 16, textureSize.x / 16};
+    if (font != FONT_PAGER) {
+        glyphOffset.y += glyphOffset.y / 4;
+    }
+    const std::vector<std::uint8_t> *glyphWidths;
+    switch (font) {
+    case FONT_PAGER:
+        glyphWidths = &fontWidthsPager;
+        break;
+    case FONT_PRICEDOWN:
+        glyphWidths = &fontWidthsPriceDown;
+        break;
+    case FONT_ARIAL:
+        glyphWidths = &fontWidthsArial;
+        break;
+    }
+    std::uint8_t monoWidth = 0;
+    if (font == FONT_PAGER) {
+        monoWidth = 1 + *std::max_element(fontWidthsPager.cbegin(),
+                                          fontWidthsPager.cend());
     }
+    fonts[font] = FontMetaData{
+        textureName,
+        *glyphWidths,
```

Comment:
I'm not happy about copying the fontWidths vector here. Does anybody have an idea about this?
Saving a pointer in the FontMetaData?